Orange/misc/environ.py

```diff
@@ -65,7 +59,13 @@ def widget_settings_dir(versioned=True):
 
     This a subdirectory of ``data_dir(versioned)`` named "widgets"
     """
-    return os.path.join(data_dir(versioned=versioned), "widgets")
+    warnings.warn(
+        f"'{__name__}.widget_settings_dir' is deprecated. "
+        "Use 'Orange.widgets.settings.widget_settings_dir'",
```

Comment:
Should this be `'Orange.canvas.config.widget_settings_dir'`?
And `'Orange.widgets.settings.widget_settings_dir'` removed since it is wrong?

Reply:
`Orange.widgets.settings.widget_settings_dir` is the effective directory (technically it is `orangewidget.settings.widget_settings_dir`.  It will *become* the correct path once the application starts and configures.

```
>>> Orange.widgets.settings.widget_settings_dir()
'/Users/aleserjavec/Library/Application Support/Orange/0.0.0/widgets'
>>> Orange.canvas.config.Config().init()
'/Users/aleserjavec/Library/Application Support/Orange/3.23.0.dev/widgets'
```
This actually has to happen very soon before the widgets are imported.

Reply:
OK, so after configuration they both return the same correct path. But still, when should one use which
`Orange.canvas.config.widget_settings_dir`
`Orange.widgets.settings.widget_settings_dir`
and could we have just one?

Currently they also have different calls to orangewidget:
`orangewidget.workflow.config.widget_settings_dir`
`orangewidget.settings.widget_settings_dir`
I think the public API of orangewidget should have one path from which to import widget_settings_dir.
If the same thing is available from multiple locations (either in orangewidget or in orange) it just creates confusion and provides opportunities for problems in the future IMO.